src/components/DataTables.js

```diff
@@ -1,19 +1,28 @@
 import React from 'react'
 import DataTable from './DataTable'
 
-export default function DataTables({datacomp, setCompIndex, setSidePlane}) {
-
+export default function DataTables({datacomp, setCompIndex, setSidePlane,setDataComp}) {
+  const remover = (keys) => {
+    setDataComp(datacomp.filter((el) => el.keys !==keys));
```

Comment:
same thing here instead of filter we can delete the element totally.

Reply:
i did filter so the component can be deleted individually

Reply:
I understand but this not removing the element from the component state. can you try creating multiple charts and see if the deleting will work

Reply:
it does